src/commcare_cloud/environment/schemas/terraform.py

```diff
@@ -58,6 +64,39 @@ class ServerConfig(jsonobject.JsonObject):
     block_device = jsonobject.ObjectProperty(lambda: BlockDevice, default=None)
     group = jsonobject.StringProperty()
     os = jsonobject.StringProperty(required=True, choices=['trusty', 'bionic', 'ubuntu_pro_bionic'])
+    count = jsonobject.IntegerProperty(default=None)
+
+    @classmethod
+    def wrap(cls, data):
+        self = super(cls, ServerConfig).wrap(data)
+        if self.count is not None and not self.server_name.split('-', 1)[0].endswith('{i}'):
+            raise ValueError('To use count, server_name must be a template string using {i}, '
+                             'and {i} must be the final part before the env suffix')
+        return self
+
+    def get_all_server_names(self):
+        if self.count is None:
+            return [self.server_name]
+        else:
+            return [self.server_name.format(i='{:03d}'.format(i)) for i in range(self.count)]
+
+    def get_all_host_names(self):
+        host_name = self.server_name.split('-', 1)[0]
+        if self.count is None:
+            return [host_name]
+        else:
+            return [host_name.format(i='{:03d}'.format(i)) for i in range(self.count)]
+
+    def get_host_group_name(self):
+        if self.count is None:
+            raise ValueError("Can only call get_host_group_name() on a server with count")
+        else:
+            return self.server_name.split('-', 1)[0][:-3]
```

Comment:
the second slice fails a sniff test for me of whether it's obvious to introspect what this code is doing without context from outside of the file. I know it trims off the last 3 characters of the string, but it's not quite obvious to me why (or what will happen to strings that don't meet this pattern).

It would be helpful to either have an example pattern available in-code to see what the code is intended to pull out, or to maybe make the code more verbose in order to more unambiguously communicate what that pattern is intended to catch.